extensions/phylum.d.ts

```diff
@@ -112,6 +112,7 @@ declare namespace Phylum {
     project?: string,
     group?: string,
     label?: string,
+    organization?: string,
```

Comment:
This is pretty ugly. `analyze(packages, project, group)` becomes `analyze(packages, project, group, null, organization)`. It seems common in ECMAScript that we'd have a `analyze(packages: (PackageWithOrigin | PurlWithOrigin)[], options: { project?: string, group?: string, label?: string, organization?: string })` form to write `analyze(packages, { project: project, group: group, organization: organization })` without having to count positional null parameters.

Organization isn't added to the documentation. I assume both group and organization are defaulted from `.phylum_project` if not specified, and then the organization is defaulted from the CLI configuration if it's not in `.phylum_project` either? Or is it that `group` and `organization` are required if `project` is specified and ignored when it's not specified (because we're using a project ID instead of a project name?)?

Reply:
I've added it to the docs. I also don't like this format, but it's the best choice for avoiding issues with existing implementations.

Reply:
The documentation doesn't specify what happens if you don't provide an organization name, and the documentation hasn't been updated for any of the other functions that got an organization parameter.

It's possible in Typescript to overload the function signature, but since it's just Javascript it doesn't actually overload the function. The callee needs to expect to receive either four optional strings or a single object, which seems like a pain with Deno op2. I guess since we're already using `#[serde]` for `packages` we could use an untagged enum to handle `project_or_options` being a string or object and then throw if `project_or_options` is the options object and any of `group` `label` `organization` are set (if we even allow passing a positional organization this way). That allows named options without breaking existing callers.

Reply:
> The documentation doesn't specify what happens if you don't provide an organization name, and the documentation hasn't been updated for any of the other functions that got an organization parameter.

It hasn't been updated for any of the other functions because none of them document parameters. I've considered just removing the parameter docs from this function too, but didn't want to do so in this PR.

Not providing an organization name is the status quo, so I don't think it needs to be explicitly documented. That's specifically the goal of these API changes.

> It's possible in Typescript to overload the function signature, but since it's just Javascript it doesn't actually overload the function.

We have basically only string arguments here. Overloading is not an option. The only viable choice that doesn't confuse **everyone** involved is adding a new parameter to the end. It might make sense to rework our API once legacy groups go away to streamline everything a bit, but I don't think it makes sense to perform drastic changes to the extension API in this patch, when there's absolutely no need to.

Reply:
> However, if we're not documenting the organization name parameter because the caller isn't supposed to provide an organization name, do we even need the organization name parameter?

We don't document any of the parameters in most other functions. Have you checked out the code of this file? We really don't need to overcomplicate things here, I'd much rather keep the changes to the extension API minimal so people can actually keep using the extensions they have.